src/openpgp.js

```diff
@@ -99,7 +99,7 @@ export function destroyWorker() {
 export function generateKey({ userIds=[], passphrase, numBits=2048, unlocked=false } = {}) {
   const options = formatUserIds({ userIds, passphrase, numBits, unlocked });
 
-  if (!util.getWebCryptoAll() && asyncProxy) { // use web worker if web crypto apis are not supported
+  if (asyncProxy) { // use web worker when available
```

Comment:
> basically for what we gain in the browser crypto, we loose with the other routines running in the main thread;

Have your tried generating a 4096 bit key? That took minutes in JS as far as I can remember. In constrast, the actual signing of keys should be constant time.

Again, I would reiterate that we should to some profiling to understand which parts of the code are the actual bottlenecks, before patching.

Thanks :)